junit-platform-commons/src/main/java/org/junit/platform/commons/util/ModuleClassFinder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import org.apiguardian.api.API;
+
+/**
+ * Class finder service providing interface.
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ * @since 1.1
+ */
+@API(status = INTERNAL, since = "1.1")
+public interface ModuleClassFinder {
+
+	/**
+	 * Special name indicating a search for all classes in all modules on the module-path.
+	 */
+	String ALL_MODULE_PATH = "ALL-MODULE-PATH";
+
+	/**
+	 * Return list of classes found in the named module that may contain testable methods.
+	 *
+	 * @param moduleName name of the module to inspect or {@code ALL-MODULE-PATH}
```

Comment:
I think having two separate methods, e.g. `findAllClassesInModule()` and `findAllClassesInModulePath()`, would be a cleaner. Thoughts?

Reply:
Having two methods makes sense here, because we have two different selectors in the launcher part. The blue-print where I took the special module name `"ALL-MODULE-PATH"` from only has a single command line option, namely: `--add-modules`. There is no `--add-all-module-path` or in our world: `--scan-module-path`.

tl;dr: Two separate methods are good - and remove the special module name `"ALL-MODULE-PATH"`.